engine/glfw/lib/android/android_window.c

```diff
@@ -326,20 +319,79 @@ void _glfwPlatformRefreshWindowParams( void )
 // Poll for new window and input events
 //========================================================================
 
-void _glfwPlatformPollEvents( void )
+
+void glfwAndroidBeginFrame()
 {
-   int ident;
-   int events;
-   struct android_poll_source* source;
+    spinlock_lock(&_glfwWinAndroid.m_RenderLock);
```

Comment:
Used to mark the entire area where the glfwBeginFrame() and the glfwPlatformSwapBuffers(), so that we don't accidentally destroy the surface during this time.
Unlocked in _glfwPlatformSwapBuffers.